extensions/vertx-http/deployment/src/test/java/io/quarkus/vertx/http/security/FormAuthNoRedirectTestCase.java

```diff
@@ -73,9 +72,8 @@ public void testFormBasedAuthSuccess() {
                 .get("/admin")
                 .then()
                 .assertThat()
-                .statusCode(302)
-                .header("location", containsString("/login"))
-                .cookie("quarkus-redirect-location", containsString("/admin"));
+                .statusCode(401)
```

Comment:
Hi @michalvavrik I'm looking at this test and I have to admit I'm not understanding what is the point of using `FormAuth` in this case, why not `Basic Auth`, if the user does not want to see the Form Auth login page, what exactly remains here from the actual FormAuth mechanics ?

Reply:
Hey Sergey, in fact I thought about same before I started looking in the issue. In the end I read argument you listed here https://quarkus.io/guides/security-built-in-authentication#basic-auth, citing:
- Credentials are sent as plain text
- Credentials are sent with each request
- Application complexity increases

Reply:
Thanks @michalvavrik, I think I'm getting it. I think this is similar to what we did for OIDC code flow, except that there we do not return 401 but 499 to let SPA intercept it...